### DISCUSSION TITLE
Type safety between resolver decorator and resolver return value

### DISCUSSION THREAD

**Discussion by miloofcroton** [MichalLytek/type-graphql#1313] [Q&A]
I'm relatively new to decorators and type-graphql, but not typescript or graphql in general.

Correct code:
```typescript
  @Query(
    () => [Post],
    { description: "Get all the posts" }
  )
  async getPosts(): Promise<Post[]> {
    return await this.items;
  }
```
No type error #1:
```typescript
  @Query(
    () => Post,
    { description: "Get all the posts" }
  )
  async getPosts(): Promise<Post[]> {
    return await this.items;
  }
```
No type error #2:
```typescript
  @Query(
    () => [Post],
    { description: "Get all the posts" }
  )
  async getPosts(): Promise<Post> {
    return await this.items.pop(); // I had to modify this code so it behaves with the asserted type Promise<Post> , which is expected
  }
```

This is a field in a class for all `Post` resolvers. It seems there's no connection between the `@Query` decorator's first param (a function with a `[Post]` return) and the return value of the function `async getPosts()`, which is `Promise<Post[]>`. If I change the value from `Post[]` to `Post`, no type error pops up.

I'm not sure if I'm using the library incorrectly or if the library just doesn't support fully type-safe resolvers with one source of truth. Can anyone shed light on this?

**Comment by MichalLytek** [comment]
See https://github.com/borremosch/eslint-plugin-type-graphql
